src/memory_pool.c

```diff
@@ -180,6 +180,14 @@ size_t umfPoolMallocUsableSize(umf_memory_pool_handle_t hPool, void *ptr) {
 
 umf_result_t umfPoolFree(umf_memory_pool_handle_t hPool, void *ptr) {
     UMF_CHECK((hPool != NULL), UMF_RESULT_ERROR_INVALID_ARGUMENT);
+    umf_memory_pool_handle_t poolTracker = umfMemoryTrackerGetPool(ptr);
```

Comment:
with this call, we would search the tracker 2 times during the free?

Reply:
AFAIK in case of Proxy Pool and Disjoint Pool - yes

Reply:
so this could significantly hurt the performance. Could you run the benchmarks and ask @lplewa for interpretation of results?

Reply:
Is this problem specific to the disjoint and proxy pools? Does the issue exist with the scalable or with the jemalloc pools?

Reply:
I think it is theoretically possible for every pool.

Reply:
As @bratpiorka mentioned it might impact performance.
so it is obvious that if an incorrect pool handle is passed by the client it is an ill-formed client's code. The question is do we want to catch such issues (and probably pay for the performance impact) or we should just say it is "undefined behavior" if an incorrect pool handle is passed as a parameter?

Reply:
I vote for UB in such case.